--- a/lib/Jifty/Plugin/OpenID.pm
+++ b/lib/Jifty/Plugin/OpenID.pm
@@ -6,7 +6,7 @@
 
 =head1 NAME
 
-Jifty::Plugin::OpenID
+Jifty::Plugin::OpenID - Provides OpenID authentication for your jifty app
 
 =head1 DESCRIPTION
 
--- a/lib/Jifty/Plugin/OpenID/Action/AuthenticateOpenID.pm
+++ b/lib/Jifty/Plugin/OpenID/Action/AuthenticateOpenID.pm
@@ -3,7 +3,7 @@
 
 =head1 NAME
 
-Jifty::Plugin::OpenID::Action::AuthenticateOpenID;
+Jifty::Plugin::OpenID::Action::AuthenticateOpenID - make OpenId authentication
 
 =cut
 
--- a/lib/Jifty/Plugin/OpenID/Action/CreateOpenIDUser.pm
+++ b/lib/Jifty/Plugin/OpenID/Action/CreateOpenIDUser.pm
@@ -3,7 +3,7 @@
 
 =head1 NAME
 
-Jifty::Plugin::OpenID::Action::CreateOpenIDUser
+Jifty::Plugin::OpenID::Action::CreateOpenIDUser - create 0penId user
 
 =cut
 
--- a/lib/Jifty/Plugin/OpenID/Action/VerifyOpenID.pm
+++ b/lib/Jifty/Plugin/OpenID/Action/VerifyOpenID.pm
@@ -3,7 +3,7 @@
 
 =head1 NAME
 
-Jifty::Plugin::OpenID::Action::VerifyOpenID;
+Jifty::Plugin::OpenID::Action::VerifyOpenID - verify OpenID
 
 =cut
 
--- a/lib/Jifty/Plugin/OpenID/Dispatcher.pm
+++ b/lib/Jifty/Plugin/OpenID/Dispatcher.pm
@@ -6,7 +6,7 @@
 
 =head1 NAME
 
-Jifty::Plugin::OpenID::Dispatcher
+Jifty::Plugin::OpenID::Dispatcher - Dispatcher for OpenID plugin
 
 =head1 DESCRIPTION
 
--- a/lib/Jifty/Plugin/OpenID/Mixin/Model/User.pm
+++ b/lib/Jifty/Plugin/OpenID/Mixin/Model/User.pm
@@ -7,7 +7,7 @@
 
 =head1 NAME
 
-Jifty::Plugin::OpenID::Mixin::Model::User
+Jifty::Plugin::OpenID::Mixin::Model::User - Mixin model user
 
 =head1 DESCRIPTION
 
--- a/lib/Jifty/Plugin/OpenID/View.pm
+++ b/lib/Jifty/Plugin/OpenID/View.pm
@@ -5,7 +5,7 @@
 
 =head1 NAME
 
-Jifty::Plugin::OpenID::View
+Jifty::Plugin::OpenID::View - Login and create pages for OpenID plugin
 
 =head1 DESCRIPTION
 
